web_src/less/_repository.less

```diff
@@ -1201,6 +1201,24 @@
         }
       }
     }
+
+    .ui.reference {
+      display: flex;
+      flex-direction: row;
+      justify-content: space-between;
+      align-items: center;
+
+      .text {
+        white-space: nowrap;
+        overflow: hidden;
+        text-overflow: ellipsis;
+        padding-right: 1rem;
+      }
+
+      .copy-button {
+        cursor: pointer;
+      }
+    }
```

Comment:
We use [Fomantic](https://fomantic-ui.com/) to handle a lot of these kind of things. Could you check if you can use the existing classes that we already have? I think most of this can be replaced with adding appropriate classes to the HTML elements.

Reply:
I've attempted to correct this with [172b5d1](https://github.com/go-gitea/gitea/pull/19609/commits/172b5d19ad8ebbc4b64f93f3e8e9154891b6d082) - problem is, I don't like how the copy button behaves at a (roughly) medium-sized resolution - it overlaps on the right side of the sidebar, making it look weird.

I think I need some assistance with this, since I'm unfamiliar with Fomantic and what I can use to emulate the same behaviour as the tailored CSS.

Rather than directly copying the style from GitLab, it might be better to directly copy some of the existing sidebar items and how they appear (with a bigger, bolder heading, and detail text underneath). Again, though, I don't know if that will make it easier to achieve less tailored CSS.

Any help would be greatly appreciated!

Reply:
So, I think this is the best non-CSS solution:
```diff
diff --git a/templates/repo/issue/view_content/sidebar.tmpl b/templates/repo/issue/view_content/sidebar.tmpl
index 1ce1deaca..14cdd6d67 100644
--- a/templates/repo/issue/view_content/sidebar.tmpl
+++ b/templates/repo/issue/view_content/sidebar.tmpl
@@ -569,12 +569,10 @@
                {{end}}
 
                <div class="ui divider"></div>
-               <div class="ui reference equal width compact grid">
-                       <div class="row">
-                               <span class="text column">References: {{.Issue.ReferenceLink}}</span>
-                               <span class="copy-button two wide column" data-clipboard-text="{{.Issue.ReferenceLink}}">
-                                       {{svg "octicon-copy"}}
-                               </span>
+               <div class="ui equal width compact grid">
+                       <div class="row ac">
+                               <span class="text column truncate">References: {{.Issue.ReferenceLink}}</span>
+                               <button class="ui two wide button column p-3" data-clipboard-text="{{.Issue.ReferenceLink}}">{{svg "octicon-copy" 14}}</button>
                        </div>
                </div>
 
diff --git a/web_src/less/_repository.less b/web_src/less/_repository.less
index cbc295f44..13f9384ba 100644
--- a/web_src/less/_repository.less
+++ b/web_src/less/_repository.less
@@ -1201,19 +1201,6 @@
         }
       }
     }
-
-    .ui.reference {
-      .text {
-        white-space: nowrap;
-        overflow: hidden;
-        text-overflow: ellipsis;
-        padding-right: 1rem;
-      }
-
-      .copy-button {
-        cursor: pointer;
-      }
-    }
   }
 
   .comment.form {
```

It uses the `truncate` for the text-overflow handling. `ac` is to make sure the text is aligned vertically. `p-3` Is to set the padding to `.5rem`. Also switched the `span` -> `button` because it's a button after all 😄. So it doesn't require any external CSS.





> I don't like how the copy button behaves at a (roughly) medium-sized resolution - it overlaps on the right side of the sidebar, making it look weird.

Well don't be ashamed, Gitea's UI isn't really perfect in terms of responsive, it's part of a greater issue. No need to do special care for now.

Reply:
OK! I just like to make sure to make it as responsive as possible (if I can).

Thanks!